src/web/app/pages-admin/admin-notifications-page/admin-notifications-page.component.spec.ts

```diff
@@ -1,30 +1,248 @@
+    fixture.detectChanges();
+
+    const btn: any = fixture.debugElement.nativeElement.querySelector('#btn-add-notification');
+    expect(btn.disabled).toBeFalsy();
+    expect(btn.textContent).toEqual(' Add New Notification');
+    btn.click();
+    fixture.detectChanges();
+
+    expect(component.isNotificationLoading).toBeFalsy();
+    expect(component.isNotificationEditFormExpanded).toBeTruthy();
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should load correct notification for a given API output', () => {
+    jest.spyOn(notificationService, 'getNotifications').mockReturnValue(of({
+      notifications: [EXAMPLE_NOTIFICATION_ONE],
+    }));
+
+    component.loadNotifications();
+
+    expect(component.notificationsTableRowModels.length).toEqual(1);
+    verifyFirstRowModelEqualToExample(EXAMPLE_NOTIFICATION_ONE);
+  });
+
+  it('should add notification for all fields filled in', () => {
+    jest.spyOn(notificationService, 'createNotification').mockReturnValue(of(EXAMPLE_NOTIFICATION_ONE));
+    const spy: SpyInstance = jest.spyOn(statusMessageService, 'showSuccessToast')
+    .mockImplementation((args: string) => {
+      expect(args).toEqual('Successfully created');
+    });
+
+    component.addNewNotificationHandler();
+
+    expect(spy).toHaveBeenCalledTimes(1);
+    expect(component.isNotificationEditFormExpanded).toBeFalsy();
+    expect(component.notificationsTableRowModels.length).toEqual(1);
+    verifyFirstRowModelEqualToExample(EXAMPLE_NOTIFICATION_ONE);
+  });
+
+  it('should display error message when failed to create notification', () => {
+    component.notificationEditFormModel = EXAMPLE_NOTIFICATION_EDIT_MODEL;
+    jest.spyOn(notificationService, 'createNotification').mockReturnValue(throwError({
+      error: {
+        message: 'This is the error message.',
+      },
+    }));
+    const spy: SpyInstance = jest.spyOn(statusMessageService, 'showErrorToast')
+      .mockImplementation((args: string) => {
+        expect(args).toEqual('This is the error message.');
+      });
+
+    component.addNewNotificationHandler();
+
+    expect(spy).toHaveBeenCalledTimes(1);
+    expect(component.notificationsTableRowModels.length).toEqual(0);
+  });
+
+  it('should display error message when notification failed to load', () => {
+    component.hasNotificationLoadingFailed = false;
+    jest.spyOn(notificationService, 'getNotifications').mockReturnValue(throwError({
+      error: {
+        message: 'This is the error message.',
+      },
+    }));
+    const spy: SpyInstance = jest.spyOn(statusMessageService, 'showErrorToast')
+      .mockImplementation((args: string) => {
+        expect(args).toEqual('This is the error message.');
+      });
+
+    component.loadNotifications();
+
+    expect(spy).toBeCalled();
+    expect(component.hasNotificationLoadingFailed).toBeTruthy();
+  });
+
+  it('should snap when notification edit form expanded for editing existent notification', () => {
+    moment.tz.setDefault('UTC');
```

Comment:
Wasn't this set in the `beforeEach` method?

Reply:
Setting it in `beforeEach` does not seem to solve the problem of difference in CI timezone, may I ask if you have any suggestions for this?

Reply:
I see. We can leave it as is then